spec/webvalve/fake_service_spec.rb

```diff
@@ -26,21 +26,44 @@ def self.name
       WebValve.reset
     end
 
-    it 'raise a useful error when an unmapped route is requested' do
-      with_env 'DUMMY_API_URL' => 'http://dummy.dev' do
-        WebValve.register subject.name
-        WebValve.setup
+    context 'when the service is at a root path' do
+      it 'raise a useful error when an unmapped route is requested' do
+        with_env 'DUMMY_API_URL' => 'http://dummy.dev' do
+          WebValve.register subject.name
+          WebValve.setup
 
-        expect { Net::HTTP.get(URI('http://dummy.dev/foos')) }.to raise_error(RuntimeError, /route not defined for GET/)
+          expect { Net::HTTP.get(URI('http://dummy.dev/foos')) }.to raise_error(RuntimeError, /route not defined for GET/)
+        end
+      end
+
+      it 'returns the result from the fake when a mapped route is requested' do
+        with_env 'DUMMY_API_URL' => 'http://dummy.dev' do
+          WebValve.register subject.name
+          WebValve.setup
+
+          expect(Net::HTTP.get(URI('http://dummy.dev/widgets'))).to eq({ result: 'it works!' }.to_json)
+        end
       end
     end
 
-    it 'returns the result from the fake when a mapped route is requested' do
-      with_env 'DUMMY_API_URL' => 'http://dummy.dev' do
-        WebValve.register subject.name
-        WebValve.setup
+    context 'when the service lives at a non-root path' do
+      it 'raise a useful error when the route is requested at the root' do
+        with_env 'DUMMY_API_URL' => 'http://dummy.dev/gg' do
+          WebValve.register subject.name
+          WebValve.setup
+
+          expect { Net::HTTP.get(URI('http://dummy.dev/widgets')) }
+            .to raise_error(WebMock::NetConnectNotAllowedError, /Real HTTP connections are disabled/)
```

Comment:
This seems to fail intermittently, I think maybe due to global state. Investigating...